src/functions.jl

```diff
@@ -0,0 +1,164 @@
+"""
+    map(f, ds, [key]; dims) -> KeyedDataset
+
+Apply function `f` to each component of the [`KeyedDataset`](@ref).
+Returns a new dataset with the same constraints, but new components.
+The function can be applied to a subselection of components via a [`Pattern`](@ref) `key` and/or `dims`.
+
+# Example
+```jldoctest
+julia> using AxisKeys, Statistics; using AxisSets: KeyedDataset;
+
+julia> ds = KeyedDataset(
+           :val1 => KeyedArray(zeros(3, 2); time=1:3, obj=[:a, :b]),
+           :val2 => KeyedArray(ones(3, 2); time=1:3, loc=[:x, :y]),
+       );
+
+julia> mean.((ds.val1, ds.val2))
+(0.0, 1.0)
+
+julia> r = map(a -> a .+ 1.0, ds; dims=:loc);
```

Comment:
It would be nice to include an example with a pattern here as well.

Reply:
That'll make this example rather long as we'll need to define a nested datasets.

Reply:
Alright, it isn't that much longer.